autoload/lsp/internal/document_hover/under_cursor.vim

```diff
@@ -28,10 +28,14 @@ function! lsp#internal#document_hover#under_cursor#do(options) abort
                 nnoremap <buffer><silent> <Plug>(lsp-float-close) :<C-u>call <SID>close_floating_window()<CR>
                 execute('doautocmd <nomodeline> User lsp_float_focused')
                 if !hasmapto('<Plug>(lsp-float-close)')
-                    imap <silent> <buffer> <C-c> <Plug>(lsp-float-close)
-                    nmap  <silent> <buffer> <C-c> <Plug>(lsp-float-close)
-                    imap <silent> <buffer> <Esc> <Plug>(lsp-float-close)
-                    nmap  <silent> <buffer> <Esc> <Plug>(lsp-float-close)
+                    if empty(mapcheck("\<C-c>", 'n'))
+                        imap <silent> <buffer> <C-c> <Plug>(lsp-float-close)
```

Comment:
we need to have if check for each line.

`imap` is for `insert` and `nmap` is for normal. so this wouldn't work.

Also might be wroth refactoring so that we have `s:register_key_bindings()` so we can reuse it.

Reply:
Oops, I'll be careful.
Fixed it to using `s:register_key_bindings`.